packages/edit-navigation/src/components/menu-editor/promise-queue.js

```diff
@@ -7,15 +7,28 @@ export default class PromiseQueue {
 		this.queue = [];
 		this.active = [];
 		this.listeners = [];
+		this.halted = false;
+	}
+
+	halt() {
+		this.halted = true;
+		this.queue = [];
+		this.listeners = [];
```

Comment:
Remember how new menu items are created whenever a new block is added? Changing an active menu should stop any further processing of these if some are in progress. My idea in this PR is to simply "halt" a promise queue which means no further actions will be processed an no listeners will be notified - eventually the entire thing will be just garbage collected.